internal/app/tfsec/checks/azu027.go

```diff
@@ -0,0 +1,99 @@
+  name                                 = "example"
+  resource_group_name                  = azurerm_resource_group.example.name
+  location                             = azurerm_resource_group.example.location
+  storage_data_lake_gen2_filesystem_id = azurerm_storage_data_lake_gen2_filesystem.example.id
+  sql_administrator_login              = "sqladminuser"
+  sql_administrator_login_password     = "H@Sh1CoR3!"
+
+  aad_admin {
+    login     = "AzureAD Admin"
+    object_id = "00000000-0000-0000-0000-000000000000"
+    tenant_id = "00000000-0000-0000-0000-000000000000"
+  }
+
+  tags = {
+    Env = "production"
+  }
+}
+`
+const AZUSynapseWorkspaceManagedNetworkGoodExample = `
+resource "azurerm_synapse_workspace" "good_example" {
+  name                                 = "example"
+  resource_group_name                  = azurerm_resource_group.example.name
+  location                             = azurerm_resource_group.example.location
+  storage_data_lake_gen2_filesystem_id = azurerm_storage_data_lake_gen2_filesystem.example.id
+  sql_administrator_login              = "sqladminuser"
+  sql_administrator_login_password     = "H@Sh1CoR3!"
+  managed_virtual_network_enabled	   = true
+  aad_admin {
+    login     = "AzureAD Admin"
+    object_id = "00000000-0000-0000-0000-000000000000"
+    tenant_id = "00000000-0000-0000-0000-000000000000"
+  }
+
+  tags = {
+    Env = "production"
+  }
+}
+`
+
+func init() {
+	scanner.RegisterCheck(scanner.Check{
+		Code: AZUSynapseWorkspaceManagedNetwork,
+		Documentation: scanner.CheckDocumentation{
+			Summary:     AZUSynapseWorkspaceManagedNetworkDescription,
+			Explanation: AZUSynapseWorkspaceManagedNetworkExplanation,
+			BadExample:  AZUSynapseWorkspaceManagedNetworkBadExample,
+			GoodExample: AZUSynapseWorkspaceManagedNetworkGoodExample,
+			Links: []string{
+				"https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/synapse_workspace#managed_virtual_network_enabled",
+				"https://docs.microsoft.com/en-us/azure/synapse-analytics/security/synapse-workspace-managed-private-endpoints",
+				"https://docs.microsoft.com/en-us/azure/synapse-analytics/security/synapse-workspace-managed-vnet",
+			},
+		},
+		Provider:       scanner.AzureProvider,
+		RequiredTypes:  []string{"resource"},
+		RequiredLabels: []string{"azurerm_synapse_workspace"},
+		CheckFunc: func(check *scanner.Check, block *parser.Block, _ *scanner.Context) []scanner.Result {
+
+			if block.MissingChild("managed_virtual_network_enabled") {
+				return []scanner.Result{
+					check.NewResult(
+						fmt.Sprintf("Resource '%s' should have managed_virtual_network_enabled set to true, the default is false.", block.FullName()),
+						block.Range(),
+						scanner.SeverityError,
+					),
+				}
+			}
+			if block.GetAttribute("managed_virtual_network_enabled").IsFalse() || block.GetAttribute("managed_virtual_network_enabled").Equals("false") || block.GetAttribute("managed_virtual_network_enabled").Equals(false) {
```

Comment:
Have you seen examples where `IsFalse()` isn't adequate? Would rather resolve that than need to have multiple `||` checking the same thing

Reply:
If the additional check is not added it will not recognize the string version of "false". Terraform sometimes allows a string in place of a bool and will not error out but using the string will pass the test without the additional test.

This would pass without the additional checks.

managed_virtual_network_enabled      = "false"

Reply:
@owenrumney This test passes for me and should fail. Please confirm your outcome.

resource "azurerm_synapse_workspace" "example" {
  name                                 = "example"
  resource_group_name                  = azurerm_resource_group.example.name
  location                             = azurerm_resource_group.example.location
  storage_data_lake_gen2_filesystem_id = azurerm_storage_data_lake_gen2_filesystem.example.id
  sql_administrator_login              = "sqladminuser"
  sql_administrator_login_password     = "H@Sh1CoR3!"
  managed_virtual_network_enabled      = "false"
  aad_admin {
    login     = "AzureAD Admin"
    object_id = "00000000-0000-0000-0000-000000000000"
    tenant_id = "00000000-0000-0000-0000-000000000000"
  }

  tags = {
    Env = "production"
  }
}